File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dic = {}
    for p in participant:
        if p not in dic:
            dic[p] = 1
        else: dic[p] += 1
        
    for c in completion:
        if dic[c] >= 1:
            dic[c] -= 1
        if dic[c] == 0:
            del dic[c]
            
    return (list(dic)[0])

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer;
    unordered_map<string, int> m;
    
    for(string s: participant){
        m[s]++;
    }
    for (string c: completion){
        m[c]--;
    }
    
    for(auto [k, v] : m){
        if (v > 0){
            answer = k;
            break;
        }
    }
    
    return answer;
}